source/fuzz/fuzzer_pass_replace_adds_subs_muls_with_carrying_extended.cpp

```diff
@@ -39,7 +39,10 @@ FuzzerPassReplaceAddsSubsMulsWithCarryingExtended::
 void FuzzerPassReplaceAddsSubsMulsWithCarryingExtended::Apply() {
   for (auto& function : *GetIRContext()->module()) {
     for (auto& block : function) {
-      for (auto& instruction : block) {
+      // Copy instructions since the transformation invalidates iterators.
+      std::vector<opt::Instruction> instr_from_block =
+          std::vector<opt::Instruction>(block.begin(), block.end());
+      for (auto& instruction : instr_from_block) {
```

Comment:
Similar to other fuzzer passes, I suggest that you instead store chosen OpIAdd/Sub/Mul instructions to a vector. Then add a second loop over the vector that gets the operand type id, finds or creates the struct type, and applies the transformation.

Reply:
@paulthomson Could you point me to a specific fuzzer pass that uses this approach?

Reply:
I think an example is FuzzerPassAddDeadBlocks